### PR TITLE
feat(metric): add recovery metrics

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR adds the following metrics related to recovery:
- recovery_failure_cnt: record the number of failure recovery attempts
- recovery_latency: record the latency of a recovery attempts

This PR also re-orders the metrics in MetaMetrics for better readability.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/barrier/recovery.rs [hunk 1]
```diff
@@ -120,86 +120,94 @@ where
...
-                let mce = self
-                    .hummock_manager
-                    .get_current_version()
-                    .await
-                    .max_committed_epoch;
-
-                if mce != INVALID_EPOCH {
-                    command_ctx.wait_epoch_commit(mce).await?;
+            let recovery_timer = self.metrics.recovery_latency.start_timer();
+            let recovery_result = {
```
Comment (yezizp2012) [inline src/meta/src/barrier/recovery.rs:124]:
```suggestion
            let recovery_result = try {
```

**Review by yezizp2012 — COMMENTED** [review]
It may be more reasonable and informative to make "recovery_latency" for the recovery of each single failover. 🤔

File: src/meta/src/barrier/recovery.rs (hunk 1 above)
Comment (hzxa21) [reply src/meta/src/barrier/recovery.rs:124]: Good catch. Fixed.

**Comment by hzxa21** [comment]
> It may be more reasonable and informative to make "recovery_latency" for the recovery of each single failover. 🤔

What do you mean by "each single failover"?

**Review by yezizp2012 — APPROVED** [review]
LGTM

**Pull request merged**